File: 240412/되돌아오기 2/come-back-2.py
moves = input()
dir_num = 0

move_dirs = {0: (0, 1), 1: (1, 0), 2: (0, -1), 3: (-1, 0)}

x, y = 0, 0
answer, times = 0, 0
for move in moves:
    if move == 'L':
        dir_num = (dir_num + 3) % 4 
    elif move == 'R':
        dir_num = (dir_num + 1) % 4
    else:
        x, y = x + move_dirs[dir_num][0], y + move_dirs[dir_num][1]
        if (x, y) == (0, 0):
            times += 1
            answer += times
            break

    if (x, y) == (0, 0):
        break

    times += 1

print(answer)

File: 260201/되돌아오기 2/come-back-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    string moves;
    cin >> moves;

    // Please write your code here.
    int dir_num = 0;
    int dx[4] = {0, 1, 0, -1};
    int dy[4] = {1, 0, -1, 0};

    long long x = 0, y = 0;
    vector<pair<long long, long long>> coords;

    long long answer = 0;
    long long times = 0;

    for (char mv : moves) 
    {
        if (mv == 'L') 
        {
            dir_num = (dir_num + 3) % 4;
        } 
        else if (mv == 'R') 
        {
            dir_num = (dir_num + 1) % 4;
        } 
        else 
        {
            x += dx[dir_num];
            y += dy[dir_num];

            if (x == 0 && y == 0) 
            {
                times += 1;
                answer += times;
                coords.push_back({x, y});
                break;
            }
            coords.push_back({x, y});
        }

        if (x == 0 && y == 0) break;
        times += 1;
    }

    bool hasOrigin = false;
    for (auto &p : coords) 
    {
        if (p.first == 0 && p.second == 0) 
        {
            hasOrigin = true;
            break;
        }
    }
    if (!hasOrigin) answer = -1;

    cout << answer << "\n";
    return 0;
}